test/query_test.rb

```diff
@@ -0,0 +1,277 @@
+  end
+
+  def test_replace
+    ref = create['ref']
+    got = client.query Fauna::Query.replace(
+      ref,
+      Fauna::Query.object(data: Fauna::Query.object(m: 1)))
+    assert_equal({ 'm' => 1 }, got['data'])
+  end
+
+  def test_delete
+    ref = create['ref']
+    client.query Fauna::Query.delete(ref)
+    assert_equal false, client.query(Fauna::Query.exists(ref))
+  end
+
+  def test_match
+    set = n_set(1)
+    assert_equal [@ref_n1, @ref_n1m1], convert_set_to_list(set)
+  end
+
+  def test_union
+    set = Fauna::Query.union n_set(1), m_set(1)
+    assert_equal [@ref_n1, @ref_m1, @ref_n1m1], convert_set_to_list(set)
+  end
+
+  def test_intersection
+    set = Fauna::Query.intersection n_set(1), m_set(1)
+    assert_equal [@ref_n1m1], convert_set_to_list(set)
+  end
+
+  def test_difference
+    set = Fauna::Query.difference n_set(1), m_set(1)
+    assert_equal [@ref_n1], convert_set_to_list(set)
+  end
+
+  def test_join
+    referenced = [create(n: 12)['ref'], create(n: 12)['ref']]
+    referencers = [create(m: referenced[0])['ref'], create(m: referenced[1])['ref']]
+
+    source = n_set 12
+    assert_equal referenced, convert_set_to_list(source)
+
+    # For each obj with n=12, get the set of elements whose data.m refers to it.
+    set = Fauna::Query.join(
+      source,
+      Fauna::Query.lambda(
+        'x',
+        Fauna::Query.match(
+          Fauna::Query.var('x'),
+          @m_index_ref)))
+    assert_equal referencers, convert_set_to_list(set)
+  end
+
+  def test_equals
+    assert_query true, Fauna::Query.equals(1, 1, 1)
+    assert_query false, Fauna::Query.equals(1, 1, 2)
+    assert_query true, Fauna::Query.equals(1)
+    assert_bad_query Fauna::Query.equals
+  end
+
+  def test_concat
+    assert_query 'abc', Fauna::Query.concat('a', 'b', 'c')
+    assert_query '', Fauna::Query.concat
+  end
+
+  def test_contains
+    obj = Fauna::Query.quote a: { b: 1 }
+    assert_query true, Fauna::Query.contains(%w(a b), obj)
+    assert_query true, Fauna::Query.contains('a', obj)
+    assert_query false, Fauna::Query.contains(%w(a c), obj)
+  end
+
+  def test_select
+    obj = Fauna::Query.quote a: { b: 1 }
+    assert_query({ 'b' => 1 }, Fauna::Query.select('a', obj))
+    assert_query 1, Fauna::Query.select(%w(a b), obj)
+    assert_query nil, Fauna::Query.select('c', obj, default: nil)
+    assert_raises(Fauna::NotFound) do
+      client.query Fauna::Query.select('c', obj)
+    end
+  end
+
+  def test_select_array
+    arr = [1, 2, 3]
+    assert_query 3, Fauna::Query.select(2, arr)
+    assert_raises(Fauna::NotFound) do
+      client.query Fauna::Query.select(3, arr)
+    end
+  end
+
+  def test_add
+    assert_query 10, Fauna::Query.add(2, 3, 5)
+    assert_bad_query Fauna::Query.add
+  end
+
+  def test_multiply
+    assert_query 30, Fauna::Query.multiply(2, 3, 5)
+    assert_bad_query Fauna::Query.multiply
+  end
+
+  def test_subtract
+    assert_query(-6, Fauna::Query.subtract(2, 3, 5))
```

Comment:
Rubocop complains because it looks like `assert_query - 6`. In this case I think it has a point.